analyses/cell-type-ewings/annotate-tumor-cells-workflow.sh

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env bash
+
+####
+: '
+This workflow will identify tumor cells in a single library.
+Tumor cell annotations are obtained by:
+
+- manually identifying cells that express marker genes
+- Running CellAssign with a list of tumor marker genes
+- Running CopyKAT to identify aneuploid cells
+
+Before running this workflow, you must download both the processed `SingleCellExperiment` objects and `AnnData` objects with `download-data.py`.
+Do this for any samples you would like to run through this workflow.
+
+To run the workflow on an individual sample provide the following arguments:
+
+sample_id: Unique sample ID (name of folder containing libray data)
+library_id: Unique library ID (prefix of data files)
+normal_celltypes: Comma separated list of cell types annotated with either `SingleR` or `CellAssign` to use as a reference list of normal cells.
+tumor_celltypes: Comma separated list of cell typs annotated with either `SingleR` or `CellAssign` that are expected to align with tumor cells.
+Any cell types used here will be used for comparing to tumor cells annotated in this workflow.
+
+Optionally, you can include the `data_dir` column. Use this if you want to use a different release other than `current` after downloading data.
+
+A set of reports summarizing the tumor cell annotations and a TSV file containing annotations from all used methods will be saved to the `results_dir`.
+The default `results_dir` is `results/annotate_tumor_cells_output/<sample_id>`.
+
+Example of running the workflow:
+
+./annotate-tumor-cells-workflow.sh \
+  --sample_id "SCPCS000490" \
+  --library_id "SCPCL000822" \
+  --normal_celltypes "Endothelial cells,endothelial cell" \
+  --tumor_celltypes "Pulmonary vascular smooth muscle cells,smooth muscle cell"
```

Comment:
I haven't done a full review, obviously, but I think we want to avoid having arguments like this to a shell script. I just don't think it is very maintainable and can easily end up with hidden bugs with misspelled option names and whatnot. The two approaches I might take would be to either convert this to a python/R script where you can use a full argument parser (or read in a sample sheet that is part of the repo), or use arrays of samples/libraries coded into the script itself and looping through those. 

The main goal is to allow a single command without any options to be used for testing, etc. That might call a script that does a sample a time, but we want a "full" test where we can.

(I might also try to avoid specifying both sample & library, if you can avoid it: if we can discover the library ids by what is present in the sample dir, that would be easier to keep track of, assuming you don't need separate different cell types for each of the libraries within a sample)

Reply:
> The two approaches I might take would be to either convert this to a python/R script where you can use a full argument parser (or read in a sample sheet that is part of the repo), or use arrays of samples/libraries coded into the script itself and looping through those.

I was trying to keep things simple in this first PR and just have a script that will run every step on a single sample so had avoided making a sample sheet. I was planning on modifying this workflow to be more formal later to be able to run on multiple samples. One option is to have a script that wraps around this and parses in the arguments, the other is to move to using snakemake and provide a sample sheet as input. I wasn't sure that either of those things were necessary yet, but I can make that change here if you prefer that.

Reply:
> I was trying to keep things simple in this first PR and just have a script that will run every step on a single sample so had avoided making a sample sheet. I was planning on modifying this workflow to be more formal later to be able to run on multiple samples. One option is to have a script that wraps around this and parses in the arguments, the other is to move to using snakemake and provide a sample sheet as input. I wasn't sure that either of those things were necessary yet, but I can make that change here if you prefer that.

I appreciate that. It's really just having the argument parsing code in bash at all in here that I am probably reacting to. Bash just isn't good at it! If you had the script use only internal variables, that would probably be fine for now. But I would try to make it work at the sample level, if that is possible.

Reply:
Okay for now I did end up adding in a sample sheet and then reading that in as the main argument of the workflow. I think we are going to ultimately want that to help track what cell types are used for which sample as normal cells, so mine as well create it now.